addons/account/models/account_tax_report.py

```diff
@@ -106,7 +106,17 @@ class AccountTaxReportLine(models.Model):
     sequence = fields.Integer(string='Sequence', required=True,
         help="Sequence determining the order of the lines in the report (smaller ones come first). This order is applied locally per section (so, children of the same line are always rendered one after the other).")
     parent_path = fields.Char(index=True, unaccent=False)
-    report_id = fields.Many2one(string="Tax Report", required=True, comodel_name='account.tax.report', ondelete='cascade', help="The parent tax report of this line")
+    report_id = fields.Many2one(
+        comodel_name='account.tax.report',
+        string="Tax Report",
+        required=True,
+        store=True,
+        readonly=False,
+        recursive=True,
+        compute='_compute_report_id',
+        ondelete='cascade',
+        help="The parent tax report of this line",
+    )
```

Comment:
@william-andre Do we still need report_id?  (and I almost bet the thing searching for the lines is SQL anyways, where you could apply the recursive over there. I doubt you did this for orm friendliness)  

It is also not forbidden to put multiple params on one line (to my knowledge).  Books have longer lines for a reason.

Reply:
> It is also not forbidden to put multiple params on one line (to my knowledge). Books have longer lines for a reason.

It should be forbidden, it is a lot less VCS friendly.

> Do we still need report_id? (and I almost bet the thing searching for the lines is SQL anyways, where you could apply the recursive over there. I doubt you did this for orm friendliness)

Doing a recursive search might be uselessly complex, but doable in SQL indeed. You would still need the column on the root lines anyway, so we might as well compute it on the child lines.
The field `line_ids` is also used in `account.tax.report` so it is somewhat useful. Just a balance of (de)normalization...